recipes/libcudss/meta.yaml

```diff
@@ -0,0 +1,198 @@
+
+{% set version = "0.3.0.9" %}
+{% set soversion = ".".join(version.split(".")[:3]) %}
+{% set somajor = version.split(".")[0] %}
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "placeholder" %}  # [osx]
+{% set extension = "tar.xz" %}  # [unix]
+{% set platform = "windows-x86_64" %}  # [win64]
+{% set extension = "zip" %}  # [win64]
+
+{% set cuda_compiler_version = cuda_compiler_version|default("None") %}
+{% set cuda_major = environ.get("cuda_compiler_version", "12.0").split(".")[0] %}
+
+package:
+  name: libcudss-split
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cudss/redist/libcudss/{{ platform }}/libcudss-{{ platform }}-{{ version }}_cuda{{ cuda_major }}-archive.{{ extension }}
+  sha256: 6eadd70e6442b43194e697c13080dae3cf1771257a334e7d7b658825a3e73a3b  # [aarch64 and (cuda_compiler_version or "").startswith("12")]
+  sha256: acc90b40334cd2e4409ec9e4200c8276f7acd50d61b7349b4ffdaeda4b2bf515  # [linux64 and (cuda_compiler_version or "").startswith("12")]
+  sha256: ac0979610d3eda9e6ce803380587118c6ec72b7dc580e6e96392ac065010196f  # [win64   and (cuda_compiler_version or "").startswith("12")]
+
+build:
+  number: 0
+  skip: true  # [cuda_compiler_version in (None, "None", "11.8")]
+  skip: true  # [not (linux64 or aarch64 or win64)]
+  script:
+    - cp -rv include $PREFIX  # [linux]
+    - cp -rv lib $PREFIX  # [linux]
+    - cp -rv include %LIBRARY_INC%  # [win]
+    - cp -rv lib %LIBRARY_LIB%  # [win]
+    - cp -rv bin %LIBRARY_BIN%  # [win]
+
+outputs:
+
+  - name: libcudss{{ somajor }}
+    build:
+      run_exports:
+        # Breaking changes every version until 1.0
+        - {{ pin_subpackage("libcudss" ~ somajor, max_pin="x.x.x") }}
+    files:
+      - lib/libcudss.so.*  # [linux]
+      - bin/cudss64_0.dll  # [win64]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cuda') }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - {{ stdlib("c") }}
+      host:
+        - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - {{ pin_subpackage("libcudss-commlayer-nccl" ~ somajor, exact=True) }}
+        - {{ pin_subpackage("libcudss-commlayer-mpi" ~ somajor, exact=True) }}
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcudss.so.{{ soversion }}  # [linux]
+        - test -L $PREFIX/lib/libcudss.so.{{ somajor }}  # [linux]
+        - test ! -f $PREFIX/lib/libcudss_static.a  # [linux]
+        - if not exist %LIBRARY_BIN%\\cudss64_{{ somajor }}.dll exit 1  # [win]
+
+  - name: libcudss-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libcudss" ~ somajor, max_pin="x.x.x") }}
+    files:
+      - lib/libcudss.so  # [linux]
+      - lib/cudss.lib  # [win]
+      - include/cudss*
```

Comment:
I treat the src/ more like "auxiiliary" than as "examples" (because we do have pure examples which show how to use the library).
Do I get it right that you suggest to do one more package so that to get these src/ the user would do `conda install libcudss, libcudss-examples` or di you mean smth else?
We can expect a significant amount of users who would want to build their own communication layer so ideally I'd have it available "by default" if it is allowed by the common practices.